sharding/simulator/service.go

```diff
@@ -79,18 +78,18 @@ func (s *Simulator) simulateNotaryRequests(fetcher mainchain.RecordFetcher, read
 			}
 
 			period := new(big.Int).Div(blockNumber.Number(), big.NewInt(s.config.PeriodLength))
+			period = period.Sub(period, big.NewInt(1))
 			req, err := syncer.RequestCollationBody(fetcher, big.NewInt(int64(s.shardID)), period)
 			if err != nil {
 				log.Errorf("Error constructing collation body request: %v", err)
 				continue
 			}
+
 			if req != nil {
-				msg := p2p.Message{
-					Peer: p2p.Peer{},
-					Data: *req,
-				}
-				s.requestFeed.Send(msg)
-				log.Info("Sent request for collation body via a shardp2p feed")
+				s.p2p.Broadcast(req)
+				log.Debug("Sent request for collation body via a shardp2p broadcast")
+			} else {
+				log.Warn("Syncer generated nil CollationBodyRequest")
```

Comment:
This is already captured by the log right before it that says "no collation found for shardid x, period y"

Reply:
syncer/handlers.go log.Debugf("No collation exists for shard")

Reply:
I see, but from this context when we ask the syncer to generate a collation request and it gives us nothing back, something feels wrong. Without debug logging the simulator appears to do nothing. 

We'll need to iterate on this to make it better.